e2e-tests/playwright/tests/functional/channels/drafts/drafts_on_deleted_message.spec.ts

```diff
@@ -0,0 +1,181 @@
+
+    // # Delete the last post by admin
+    try {
+        await adminClient.deletePost(adminPost.id);
+    } catch (error) {
+        throw new Error('Failed to delete post by admin');
+    }
+
+    // * Verify drafts in user's textbox is still visible
+    const rhsTextboxValue = await sidebarRight.getTextboxValue();
+    expect(rhsTextboxValue).toBe(draftMessageByUser);
+
+    // # Click on remove post
+    const deletedPostByAdminInRHS = await sidebarRight.getRHSPostById(adminPost.id);
+    await deletedPostByAdminInRHS.clickOnRemovePost();
+
+    // * Verify the drafts links should also be removed from sidebar
+    await channelPage.sidebarLeft.draftsDoesntExist();
+});
+
+test('MM-T5435_2 Gloabl Drafts link in sidebar should be hidden when user deletes root post ', async ({pw, pages}) => {
+    const {team, user, userClient} = await pw.initSetup();
+
+    try {
+        // # Set preferences to disable tooltips for drafts and threads
+        await userClient.savePreferences(user.id, createNoTooltipForDraftsAndThreadsPreferences(user.id));
+    } catch (error) {
+        throw new Error('Failed to set preferences');
+    }
+
+    // # Log in as user in new browser context
+    const {page} = await pw.testBrowser.login(user);
+
+    // # Visit default channel page
+    const channelPage = new pages.ChannelsPage(page);
+    await channelPage.goto(team.name);
+    await channelPage.toBeVisible();
+
+    // # Post a message in the channel
+    await channelPage.postMessage('Message which will be deleted');
+    await channelPage.sendMessage();
+
+    // # Start a thread by clicking on reply button
+    const lastPost = await channelPage.getLastPost();
+    await lastPost.openRHSWithPostOptions();
+
+    const sidebarRight = channelPage.sidebarRight;
+    await sidebarRight.toBeVisible();
+
+    // # Write a message in the thread
+    await sidebarRight.postMessage('Replying to a thread');
+    await sidebarRight.sendMessage();
+
+    // # Write a message in the reply thread but don't send it
+    await sidebarRight.postMessage('I should be in drafts');
+
+    // # Bring focus back to center channel textbox, so that the draft in RHS is saved
+    await channelPage.postMessage('');
+
+    // * Verify drafts link in channel sidebar is visible
+    await channelPage.sidebarLeft.draftsExist();
+
+    // # Delete the last post with post options
+    await lastPost.openPostActionsMenu();
+    const postOptionsMenuOnLastPost = channelPage.postActionMenu;
+    await postOptionsMenuOnLastPost.toBeVisible();
+    await postOptionsMenuOnLastPost.click('Delete', true);
+    const deletePostModal = channelPage.deletePostModal;
+    await deletePostModal.toBeVisible();
+    await deletePostModal.confirmClick();
+
+    // * Verify drafts link in channel sidebar is visible
+    await channelPage.sidebarLeft.draftsDoesntExist();
+});
+
+function createNoTooltipForDraftsAndThreadsPreferences(userId: UserProfile['id']): PreferenceType[] {
```

Comment:
This may not be needed here once move disabling preference in a global location.

Reply:
Done ✅